neuraxle/base.py

```diff
@@ -1556,7 +1727,65 @@ def _sklearn_to_neuraxle_step(step) -> BaseStep:
     return step
 
 
-class MetaStepMixin:
+class _HasChildrenMixin:
+    """
+    Mixin to add behavior to the steps that have children (sub steps).
+
+    .. seealso::
+        :class:`~neuraxle.base.MetaStepMixin`,
+        :class:`~neuraxle.base.TruncableSteps`,
+        :class:`~neuraxle.base.TruncableSteps`
+    """
+    def apply(self, method: Union[str, Callable], ra: _RecursiveArguments = None, *args, **kwargs) -> RecursiveDict:
+        """
+        Apply method to root, and children steps.
+        Split the root, and children values inside the arguments of type RecursiveDict.
+
+        :param method: str or callable function to apply
+        :param ra: recursive arguments
+        :return:
+        """
+        ra: _RecursiveArguments = _RecursiveArguments(ra=ra, *args, **kwargs)
+        results: RecursiveDict = self._apply_self(method=method, ra=ra)
+        results: RecursiveDict = self._apply_childrens(results=results, method=method, ra=ra)
+
+        return results
+
+    def _apply_self(self, method: Union[str, Callable], ra: _RecursiveArguments):
+        terminal_ra: _RecursiveArguments = ra[None]
+        self_results: RecursiveDict = BaseStep.apply(self, method=method, ra=terminal_ra)
+        self_results: RecursiveDict = self._purge_apply_results(self_results)
+        return self_results
+
+    def _apply_childrens(self, results: RecursiveDict, method: Union[str, Callable], ra: _RecursiveArguments) -> RecursiveDict:
+        for children in self.get_children():
+            children_results = children.apply(method=method, ra=ra[children.get_name()])
+
+            children_results = self._purge_apply_results(children_results)
+            results[children.get_name()] = children_results
```

Comment:
If children_results is empty, some code inside `__setitem__` it should ignore the provided children_results.

Reply:
Note: this is a reminder, it might already work like this.